armory/scenarios/poisoning_witches_brew.py

```diff
@@ -0,0 +1,260 @@
+                "Not computing fairness metrics.  If these are desired, set 'compute_fairness_metrics':true under the 'adhoc' section of the config"
+            )
+
+    def load_dataset(self, eval_split_default="test"):
+        # Over-ridden because we need batch_size = 1 for the test set for this attack.
+
+        dataset_config = self.config["dataset"]
+        dataset_config = copy.deepcopy(dataset_config)
+        dataset_config["batch_size"] = 1
+        eval_split = dataset_config.get("eval_split", eval_split_default)
+        log.info(f"Loading test dataset {dataset_config['name']}...")
+        self.test_dataset = config_loading.load_dataset(
+            dataset_config,
+            split=eval_split,
+            num_batches=self.num_eval_batches,
+            **self.dataset_kwargs,
+        )
+        self.i = -1
+
+    def run_benign(self):
+        # Called for all non-triggers
+
+        x, y = self.x, self.y
+
+        x.flags.writeable = False
+        y_pred = self.model.predict(x, **self.predict_kwargs)
+
+        self.non_trigger_accuracy_metric.add_results(y, y_pred)
+
+        for y_, y_pred_ in zip(y, y_pred):
+            if y_ not in self.benign_test_accuracy_per_class.keys():
+                self.benign_test_accuracy_per_class[y_] = []
+
+            self.benign_test_accuracy_per_class[y_].append(
+                y_ == np.argmax(y_pred_, axis=-1)
+            )
+
+        self.y_pred = y_pred  # for exporting when function returns
+
+    def run_attack(self):
+        # Only called for the trigger images
+
+        x, y = self.x, self.y
+
+        x.flags.writeable = False
+        y_pred_adv = self.model.predict(x, **self.predict_kwargs)
+
+        self.trigger_accuracy_metric.add_results(y, y_pred_adv)
+
+        self.y_pred_adv = y_pred_adv  # for exporting when function returns
+
+    def evaluate_current(self):
+
+        if self.i in self.trigger_index:
+            self.run_attack()
+        else:
+            self.run_benign()
+
+        # TODO Needs discussion--what is useful to export.
+        # This just exports clean test samples up to num_eval_batches, and all the triggers.
+        # The use of sample_exporter is nonconventional since in this attack,
+        # we don't have benign and adversarial versions of the same test image.
+        if (
+            self.num_export_batches > self.sample_exporter.saved_batches
+        ) or self.i in self.trigger_index:
+            if self.sample_exporter.saved_samples == 0:
+                self.sample_exporter._make_output_dir()
+            name = "trigger" if self.i in self.trigger_index else "non-trigger"
+            self.sample_exporter._export_image(self.x[0], name=name)
+            self.sample_exporter.saved_samples += 1
+            self.sample_exporter.saved_batches = (
+                self.sample_exporter.saved_samples
+                // self.config["dataset"]["batch_size"]
+            )
+
+    def _add_accuracy_metrics_results(self):
```

Comment:
Need to call `super()._add_accuracy_metrics_results()` if you want those results too.

Reply:
We don't; the original metrics were not relevant for this attack.